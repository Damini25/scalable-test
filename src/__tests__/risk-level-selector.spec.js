import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import InputSelector from '../risk-level-selector';

const props = {
    riskLevel: 3,
    years: 1,
    investmentSum: 1000,
    monthlySum: 100,
    onInputchange:jest.fn(),
    riskOptions:[3,4,5,6,7]
}
afterEach(cleanup)
test('render <InputSelector>', () => {
    const wrapper = render(<Router><InputSelector {...props}/></Router>)
    expect(wrapper).toMatchSnapshot()
})
test('function call on risk options change',()=>{
    const wrapper = render(<Router><InputSelector {...props}/></Router>)
    const inp = document.querySelector("[data-testid=risk-options-id]");
    const event = {
        target: { value: 3 }
      }
    fireEvent.change(inp,event)
    expect(props.onInputchange).toHaveBeenCalledTimes(1)
})
test('function call on years options change',()=>{
    render(<Router><InputSelector {...props}/></Router>)
    const inp = document.querySelector("[data-testid=years-id]");
    const event = {
        target: { value: '3' }
      }
    fireEvent.change(inp,event)
    expect(props.onInputchange).toHaveBeenCalledTimes(1)
})