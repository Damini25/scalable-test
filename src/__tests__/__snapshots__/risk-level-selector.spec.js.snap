// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <InputSelector> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <label>
            Risk level:
          </label>
          <select
            data-testid="risk-options-id"
            name="riskLevel"
          >
            3
            4
            5
            6
            7
          </select>
        </div>
        <div>
          <label>
            No. of years:
          </label>
          <input
            data-testid="years-id"
            max="1000"
            min="1"
            name="years"
            type="number"
            value="1"
          />
        </div>
        <div>
          <label>
            Initial Investment Sum:
          </label>
          <input
            max="10000"
            min="1"
            name="investmentSum"
            type="number"
            value="1000"
          />
        </div>
        <div>
          <label>
            Initial Monthly Sum:
          </label>
          <input
            max="10000"
            min="1"
            name="monthlySum"
            type="number"
            value="100"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <label>
          Risk level:
        </label>
        <select
          data-testid="risk-options-id"
          name="riskLevel"
        >
          3
          4
          5
          6
          7
        </select>
      </div>
      <div>
        <label>
          No. of years:
        </label>
        <input
          data-testid="years-id"
          max="1000"
          min="1"
          name="years"
          type="number"
          value="1"
        />
      </div>
      <div>
        <label>
          Initial Investment Sum:
        </label>
        <input
          max="10000"
          min="1"
          name="investmentSum"
          type="number"
          value="1000"
        />
      </div>
      <div>
        <label>
          Initial Monthly Sum:
        </label>
        <input
          max="10000"
          min="1"
          name="monthlySum"
          type="number"
          value="100"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
